### ISSUE TITLE
o-my-posh theme slightly broken

### ISSUE THREAD

**Issue by imSamiul** [microsoft/cascadia-code#766]
### Cascadia family version

2404.23

### Cascadia family variant(s)

Cascadia Code (the version with ligatures)

### Font file format(s)

.ttf (variable)

### Platform

Windows

### Other Software

Windows Terminal

### What happened?

![Screenshot 2024-09-02 135205](https://github.com/user-attachments/assets/24d08ef2-d1ca-4827-85a9-d78a5551bf68)
Here o-my-posh theme icon is slightly broken on new released Cascadia Code 2404.23 Nerd Font.

**Comment by o-l-a-v** [comment]
Do you have an example of a font where it's not broken like that?

**Comment by imSamiul** [comment]
> Do you have an example of a font where it's not broken like that?
 Fira Code Nerd Font Mono
 
![image](https://github.com/user-attachments/assets/145a22d1-4718-4daf-8361-c9c78398ae68)

@o-l-a-v

**Comment by o-l-a-v** [comment]
I can't reproduce. Tried Cascadia Mono NF and Cascadia Code NF.

<img width="565" alt="image" src="https://github.com/user-attachments/assets/29f6a81d-3e1c-450f-8bb5-252067c0341d">

**Comment by imSamiul** [comment]
@o-l-a-v  it gets fixed when the font size increased. Try using lower the font size of windows terminal then use get-PoshThemes. some themes are broken.

**Comment by o-l-a-v** [comment]
I used font size 12 in those screenshots and zoomed in. I also tried `Get-PoshThemes`. I can't reproduce with the steps you've provided. So maybe you should do some more research into what conditions causes this.

**Comment by aaronbell** [comment]
Not to say that there isn’t something else going on here, but it is challenging to identify exactly the problem due to the differing factors involved. Namely the way that a given rendering environment implements the highlighting, the way the OS text rasterizer positions the end cap, the resolution of your monitor, zoom amount, etc etc.

I think that the next version of Cascadia will sort this out but hard to say for certain.

**Comment by imSamiul** [comment]
My problem is kind of wierd. My themes get broken only on even number of font size like 10,12,14... 16 is good. don't know what happed. 11,13,15 are good. hard to find out the problem. I use windows 11, my laptop resolution is 1920x1080 with 125% zoom 15.6 inch display. maybe it pixer rendering issue. not sure. any information that you need i can provide. Hope for the better version of Cascadia.

**Comment by aaronbell** [comment]
Yeah, that’s kind of how this thing goes. Some sizes are perfect, others aren’t. Hard to predict!

**Comment by DHowett** [comment]
It is worth noting that new versions of Windows Terminal (1.20 and above) take over the following powerline shapes, and draw them "perfectly". This means that if you are using Cascadia with these versions Windows Terminal, you should not experience rounding errors like this.



![Image](https://github.com/user-attachments/assets/6710fc48-4b64-4f87-ba99-20937870d3b5)
